base/pkg/entry.jl

```diff
@@ -143,23 +162,43 @@ function status(io::IO, pkg::AbstractString, ver::VersionNumber, fix::Bool)
     fix || return println(io,ver)
     @printf io "%-19s" ver
     if ispath(pkg,".git")
-        print(io, Git.attached(dir=pkg) ? Git.branch(dir=pkg) : Git.head(dir=pkg)[1:8])
-        attrs = AbstractString[]
-        isfile("METADATA",pkg,"url") || push!(attrs,"unregistered")
-        Git.dirty(dir=pkg) && push!(attrs,"dirty")
-        isempty(attrs) || print(io, " (",join(attrs,", "),")")
+        prepo = GitRepo(pkg)
+        try
+            with(LibGit2.head(prepo)) do phead
+                if LibGit2.isattached(prepo)
+                    print(io, LibGit2.shortname(phead))
+                else
+                    print(io, string(LibGit2.Oid(phead))[1:8])
+                end
+            end
+            attrs = AbstractString[]
+            isfile("METADATA",pkg,"url") || push!(attrs,"unregistered")
+            LibGit2.isdirty(prepo) && push!(attrs,"dirty")
+            isempty(attrs) || print(io, " (",join(attrs,", "),")")
+        catch
+            print(io, "broken-repo (unregistered)")
+        finally
+            finalize(prepo)
+        end
     else
         print(io, "non-repo (unregistered)")
     end
     println(io)
 end
 
+function status(io::IO, pkg::AbstractString, msg::AbstractString)
+    @printf io " - %-29s " pkg
+    @printf io "%-19s" msg
+    println(io)
+end
```

Comment:
Can't all three lines be combined into a single `@printf io " - %-29s %-19s\n" pkg msg`?